File: SandBank.API/Models/UserActivity.cs
using System.ComponentModel.DataAnnotations;

namespace SandBank.Models;

public class UserActivity
{
    public int Id { get; set; }

    [Required]
    public int UserId { get; set; }

    [Required]
    public int ActivityId { get; set; }

    [Required]
    public string Status { get; set; } = string.Empty;

    [Required]
    public string Role { get; set; } = string.Empty;

    [Required]
    public DateTime AddedAt { get; set; }

    // Relations — nullable, no [Required]
    public User? User { get; set; }
    public Activity? Activity { get; set; }
}

File: SandBank.API/models/user_activity.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from database import Base

class UserActivity(Base):
    __tablename__ = "user_activities"

    id = Column(Integer, primary_key=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    activity_id = Column(Integer, ForeignKey("activities.id"), nullable=False)
    status = Column(String, nullable=False)
    role = Column(String, nullable=False)
    added_at = Column(DateTime, nullable=False)

    user = relationship("User", back_populates="user_activities")